src/lib.rs

```diff
@@ -224,13 +225,71 @@ fn fd_cloexec(fd: u32) -> io::Result<()> {
     Ok(())
 }
 
+/// Asks the service manager for enabled watchdog.
+///
+/// If the `unset_env` parameter is set, the `WATCHDOG_USEC` and `WATCHDOG_PID` environment variables
+/// will be unset before returning. Further calls to `watchdog_enabled` will fail, but
+/// child processes will no longer inherit the variable.
+///
+/// See [`sd_watchdog_enabled(3)`][sd_watchdog_enabled] for details.
+///
+/// [sd_watchdog_enabled]: https://www.freedesktop.org/software/systemd/man/sd_watchdog_enabled.html
+///
+///
+/// # Example
+///
+/// ```no_run
+/// # use sd_notify;
+/// #
+/// let mut usec;
+/// let enabled = sd_notify::watchdog_enabled(true, &mut usec);
+/// ```
+pub fn watchdog_enabled(unset_env: bool, usec: &mut u64) -> io::Result<bool> {
+    let s = env::var_os("WATCHDOG_USEC");
+    let p = env::var_os("WATCHDOG_PID");
+
+    let finish = |res: io::Result<bool>| -> io::Result<bool> {
+        if unset_env {
+            env::remove_var("WATCHDOG_USEC");
+            env::remove_var("WATCHDOG_PID");
+        }
+
+        res
+    };
+
+    if let Some(t) = s
+        .to_owned()
+        .and_then(|s| s.to_str().and_then(|s| u64::from_str(s).ok()))
+    {
+        if t <= 0 || t >= u64::MAX {
```

Comment:
:wave: 

This can't actually happen because `t` is already an `u64`. If it's invalid, you'll end up on the (missing) `else` branch.

Reply:
You're right. I copied the check from the original C API.